packages/cactus-common/src/test/typescript/unit/js-object-signer.test.ts

```diff
@@ -0,0 +1,210 @@
+// tslint:disable-next-line: no-var-requires
+import test from "tape";
+
+import {
+  JsObjectSigner,
+  IJsObjectSignerOptions,
+} from "../../../main/typescript/js-object-signer";
+
+import { Secp256k1Keys } from "../../../main/typescript/signer-key-pairs";
+
+import crypto from "crypto";
+import secp256k1 from "secp256k1";
+import stringify from "json-stable-stringify";
+
+const keyPairs = Secp256k1Keys.generateKeyPairsBuffer();
+
+const hashFunction = (data: any): string => {
+  return crypto.createHash("sha256").update(stringify(data)).digest("hex");
+};
+
+const signFunction = (msg: any, pkey: any): any => {
+  const signObj = secp256k1.ecdsaSign(
+    Buffer.from(hashFunction(msg), `hex`),
+    Buffer.from(pkey, `hex`)
+  );
+  return signObj.signature;
+};
+
+const verifySignFunction = (
+  msg: any,
+  signature: any,
+  pubKey: Uint8Array
+): boolean => {
+  return secp256k1.ecdsaVerify(
+    signature,
+    Buffer.from(hashFunction(msg), `hex`),
+    pubKey
+  );
+};
+
+test("Simple JSON Test", async (assert: any) => {
+  const jsObjectSignerOptions: IJsObjectSignerOptions = {
+    privateKey: keyPairs.privateKey,
+    logLevel: "debug",
+  };
+  const jsObjectSigner = new JsObjectSigner(jsObjectSignerOptions);
+
+  const payload1 = { field1: "test11", field2: "test12", field3: 13 };
+  const sign1 = jsObjectSigner.sign(payload1);
+
+  const payload2 = { field3: 13, field2: "test12", field1: "test11" };
+  const sign2 = jsObjectSigner.sign(payload2);
+
+  assert.equals(sign1.toString, sign2.toString);
+});
+
+test("Simple Nested JSON Test", async (assert: any) => {
+  const jsObjectSignerOptions: IJsObjectSignerOptions = {
+    privateKey: keyPairs.privateKey,
+    logLevel: "debug",
+  };
+  const jsObjectSigner = new JsObjectSigner(jsObjectSignerOptions);
+
+  const inner1 = { someProperty: "cool", otherStuff: "also cool" };
+  const outer1 = { innerProperty: inner1, outerProperty: "test" };
+  const sign1 = jsObjectSigner.sign(outer1);
+
+  const inner2 = { otherStuff: "also cool", someProperty: "cool" };
+  const outer2 = { outerProperty: "test", innerProperty: inner2 };
+  const sign2 = jsObjectSigner.sign(outer2);
+
+  assert.equals(sign1.toString, sign2.toString);
+});
+
+test("Simple Date JSON Test", async (assert: any) => {
```

Comment:
The `assert` parameter can be typed by importing the `Test` type from the `tape` library using the brackets syntax like this:
```typescript
import test, { Test } from "tape";
```

If you do this then the IDE will have auto complete for all the possible assertion methods that the testing library provides (big help when you don't have to look up stuff like how do they call the deepEquals assertion method and other things like that.